### PR TITLE
test : nodeAddressForLB()

### PR DESCRIPTION
**Pull request by KingDaemonX**
**What this PR does / why we need it**:

**Which issue this PR fixes(if applicable)**:
fixes #

**Special notes for reviewers**:

**Release note**:

```release-note
NONE
```
implemented unitest for all possible edge case that this function can encounter checking in strategic range

### CODE REVIEW

File: pkg/openstack/loadbalancer_test.go
Comment (dulek) [inline pkg/openstack/loadbalancer_test.go:823]: Please make sure to use `assert.Equal()` and `assert.*Error*()` methods instead of manual comparisons.

**Review by dulek — CHANGES_REQUESTED** [review]
Wow, nice work, extensive number of cases! Please just fix the problem with manual comparisons instead of asserts and I think this is good to go.

**Review by dulek — CHANGES_REQUESTED** [review]

File: pkg/openstack/loadbalancer_test.go
Comment (KingDaemonX) [reply pkg/openstack/loadbalancer_test.go:819]: noted !
i already made the mod

**Review by dulek — CHANGES_REQUESTED** [review]
Ah, I've just noticed this, in some cases you should throw in some other address types to make sure they're ignored [1], e.g. `NodeExternalDNS`.

[1] https://github.com/kubernetes/cloud-provider-openstack/blob/3cbb728ba80b15a7cc66b9a936bfa78cb62b5c9c/pkg/openstack/loadbalancer.go#L611

**Comment by KingDaemonX** [comment]
@pierreprinetti if you have the time, this is kinda good to go and just needs a /lgtm tag

**Review by pierreprinetti — APPROVED** [review]
LGTM! Thank you for your contribution.

**Comment by pierreprinetti** [comment]
@jichenjc @kayrus PTAL

**Comment by jichenjc** [comment]
/approve
/hold 

seems some conflict need to be solved? let's solve this then unhold

**Comment by KingDaemonX** [comment]
@jichenjc i just rebased and ready to check and go

**Comment by KingDaemonX** [comment]
i think it needs a /lgtm to also be merged @jichenjc

**Comment by KingDaemonX** [comment]
@pierreprinetti thank you very much :)

**Pull request merged**
